consumption.tex

```diff
@@ -0,0 +1,50 @@
+\section{Consumption}
+
+We describe how the target blockchain can verify a burn which happened on the source blockchain. Specifically, we will study the case of Bitcoin and Ethereum as source blockchains. Most cryptocurrencies on the market are adaptations of either of these. We define a party which we call a \emph{prover} who is responsible for constructing a proof of the event and sending it to the target blockchain. The proof we need to send is one of transaction inclusion in the honestly adopted chain of the source blockchain. We will denote this proof as a triple $\beta = \sf(txRaw, txInclusion, blockHeader, blockConnection)$.
+
+\subsection{Transaction Inclusion Proof}
+
+For simplicity we will abbreviate the transaction inclusion part of the complete proof as $\alpha = \sf(txRaw, txInclusion, blockHeader)$.
+
+\import{./}{algorithms/alg.verify-tx.tex}
+
+The transaction inclusion proof is conveniently defined by both Bitcoin and Ethereum. In Bitcoin, each block header contains a commitment to the set of the transaction ids in the block in the form of a Merkle Tree root. Thus, for Bitcoin our transaction inclusion proof is of the form $\mathsf{txInclusion} = (\mathsf{txIDMerkleRoot}, \mathsf{txIDMerkleProof})$. The shape of the proof is as follows $\mathsf{txIDMerkleProof} = ([x_0, x_1, \dots], \mathsf{txIndex}, \mathsf{txCount})$.
+
+Ethereum stores a similar commitment in its header, in the form of the root of a Patricia-Merkle Trie. This trie contains the transactions of the block, using the transaction id of each transaction as a key to determine the path to follow starting from the root in order to arrive at that transaction. Similar to regular Merkle trees, Patricia-Merkle tries allow for leaf inclusion proofs. Thus the form of \textsf{txInclusion} is the same as for Bitcoin transaction inclusion proofs. However, the shape of the merkle proof is different: $\mathsf{txIDMerkleProof} = [[x_{1,1}, x_{1,2}, \dots], \dots]$.
+
+The verification algorithm for $\alpha$ is presented in Algorithm~\ref{alg.verify-tx}. At the heart of the verification is the function \textsf{containsTransfer}, which takes a transaction, an address and an amount and determines if a value transfer of that exact amount to the address provided has ocured inside the transaction provided. For Bitcoin, this function locates a P2PKH output with the exact amount and receiving address. For Ethereum, the function ensures that the transaction comprises a fund transfer (i.e. the \textsf{data} field is empty), to the receiving address provided and with the exact amount. If this check passes and the transaction inclusion proofs are valid, \textsf{verify-tx} function accepts the proof $\alpha$ for the specified event $e$.
+
+\subsection{Block Connection Proof}
+
+We will look at two ways of proving block connection to the honestly adopted source chain.
```

Comment:
Avoid future tense

Reply:
What is it that we're trying to do here? Is the reader familiar with what we mean by "block connection"? I don't think so.

Reply:
Perhaps it's best to have a sentence which articulates the two ways before we explain them.